third_party/move/move-model/src/demove_helper.rs

```diff
@@ -0,0 +1,133 @@
+use std::{collections::BTreeMap, rc::Rc, borrow::Borrow};
+
+use codespan::Files;
+use itertools::Itertools;
+use move_command_line_common::files::FileHash;
+use move_compiler::expansion::ast::Program;
+
+use crate::{
+    ast::*, builder::{module_builder::ModuleBuilder, model_builder::ModelBuilder}, expansion_script_to_module, model::*, symbol::*,
+};
+
+impl ModuleData {
+    pub fn dummy(name: ModuleName, id: usize) -> ModuleData {
+        ModuleData {
+            name,
+            id: ModuleId::new(id),
+            attributes: Default::default(),
+            compiled_module: None,
+            source_map: None,
+            named_constants: Default::default(),
+            struct_data: Default::default(),
+            struct_idx_to_id: Default::default(),
+            function_data: Default::default(),
+            function_idx_to_id: Default::default(),
+            spec_vars: Default::default(),
+            spec_funs: Default::default(),
+            module_spec: Default::default(),
+            loc: Default::default(),
+            spec_block_infos: Default::default(),
+            used_modules: Default::default(),
+            used_modules_including_specs: Default::default(),
+            friend_modules: Default::default(),
+            use_decls: Vec::new(),
+            friend_decls: Vec::new(),
+        }
+    }
+}
+
+impl StructData {
+    /// Creates a new struct data.
+    pub fn dummy(name: Symbol) -> StructData {
+        StructData {
+            name,
+            loc: Default::default(),
+            abilities: AbilitySet::EMPTY,
+            def_idx: None,
+            attributes: Default::default(),
+            type_params: Default::default(),
+            spec_var_opt: Default::default(),
+            field_data: Default::default(),
+            spec: Default::default(),
+        }
+    }
+}
+
+impl FunctionData {
+    /// Creates a new function data.
+    pub fn dummy(name: Symbol) -> FunctionData {
+        FunctionData {
+            name,
+            loc: Default::default(),
+            def_idx: Default::default(),
+            handle_idx: Default::default(),
+            visibility: Default::default(),
+            is_native: Default::default(),
+            kind: FunctionKind::Regular,
+            attributes: Default::default(),
+            type_params: Default::default(),
+            params: Default::default(),
+            result_type: crate::ty::Type::Error,
+            access_specifiers: Default::default(),
+            spec: Default::default(),
+            def: Default::default(),
+            called_funs: Default::default(),
+            calling_funs: Default::default(),
+            transitive_closure_of_called_funs: Default::default(),
+        }
+    }
+}
+
+pub fn run_stackless_compiler(env: &mut GlobalEnv, program: Program) {
```

Comment:
If you want to compile from sources without generating bytecode, please use the compiler v2 API here: https://github.com/aptos-labs/aptos-core/blob/19704a3a0d4e4ad0941125e6a1c9256f7abfdc38/third_party/move/move-compiler-v2/src/lib.rs#L119-L118.

Otherwise lets chat how we can extend the v2 API if it does not work for you.

Notice that you are kind of indirectly already calling the v2 compiler the way how you are doing things here.

Reply:
Please address this.